seqr/views/apis/anvil_workspace_api.py

```diff
@@ -1,57 +1,133 @@
-    #  just be the workspace name. Make sure to set workspace_namespace and workspace_name correctly;
-    # 5) Add families/individuals based on the uploaded pedigree file;
-    # 6) Send an email to all seqr data managers saying a new AnVIL project is ready for loading. Include the seqr
-    #  project guid, the workspace name, and attach a txt file with a list of the individual IDs that were created.
-    return create_json_response(response_json)
+    project_name = _get_project_name(namespace, name)
+    project = Project.objects.filter(name = project_name)
+    if project:
+        error = 'Project {} exists.'.format(project_name)
+        logger.warning(error)
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Validate that the current user has logged in through google and has sufficient permissions
+    if not workspace_has_perm(request.user, CAN_EDIT, namespace, name, can_share=True):
+        error = "Missing required permissions for loading data from workspace {}/{}".format(namespace, name)
+        logger.warning(error)
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Validate all the user inputs from the post body
+    request_json = json.loads(request.body)
+
+    missing_fields = [field for field in ['genomeVersion', 'uploadedFileId'] if not request_json.get(field)]
+    if missing_fields:
+        error = 'Field(s) "{}" are required'.format(', '.join(missing_fields))
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    if not request_json.get('agreeSeqrAccess'):
+        error = 'Must agree to grant seqr access to the data in the associated workspace.'
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Add the seqr service account to the corresponding AnVIL workspace
+    add_service_account(request.user, namespace, name)
+
+    # Parse families/individuals in the uploaded pedigree file
+    try:
+        json_records = load_uploaded_file(request_json['uploadedFileId'])
+    except Exception as ee:
+        error = "Uploaded pedigree file is missing or other exception: {}".format(str(ee))
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    pedigree_records, errors, ped_warnings = parse_pedigree_table(json_records, 'ped_file', user=request.user, project=project)
+    if errors:
+        error = "Parse pedigree data failed. {}".format(errors)
+        return create_json_response({'error': error}, status=400, reason=error)
+
+    # Create a new Project in seqr
+    project_args = {
+        'name': project_name,
+        'genome_version': request_json['genomeVersion'],
+        'description': request_json.get('description', ''),
+        'workspace_namespace': namespace,
+        'workspace_name': name,
+    }
+
+    project = create_model_from_json(Project, project_args, user = request.user)
```

Comment:
nitpicky, but you have a lot of spaces around equal signs in this PR. I really recommend you change the settings in whatever you are using for code editing so this stops happening

Reply:
I reconfigure `PyCharm` to have no space around `=` in named parameters and key. Hope these spaces will go away in the future.